lib/matplotlib/contour.py

```diff
@@ -1513,39 +1505,34 @@ def _check_xyz(self, args, kwargs):
         z = ma.asarray(args[2], dtype=np.float64)
 
         if z.ndim != 2:
-            raise TypeError("Input z must be a 2D array.")
-        elif z.shape[0] < 2 or z.shape[1] < 2:
-            raise TypeError("Input z must be at least a 2x2 array.")
-        else:
-            Ny, Nx = z.shape
+            raise TypeError(f"Input z must be 2D, not {z.ndim}D")
+        if z.shape[0] < 2 or z.shape[1] < 2:
+            raise TypeError(f"Input z must be at least a (2, 2) array, but "
+                            f"has shape {z.shape}")
+        Ny, Nx = z.shape
 
         if x.ndim != y.ndim:
-            raise TypeError("Number of dimensions of x and y should match.")
-
+            raise TypeError(f"Dimensionalities of x ({x.ndim}) and y "
```

Comment:
I think I prefer "Number of dimensions" here (I didn't know what Dimensionalities meant until I looked at the before diff)

Reply:
edited